functions/self_update.go

```diff
@@ -0,0 +1,40 @@
+package functions
+
+import (
+	"log"
+	"strings"
+
+	"github.com/blang/semver"
+	"github.com/gravitl/netclient/daemon"
+	"github.com/rhysd/go-github-selfupdate/selfupdate"
+)
+
+func SelfUpdate(currentVersion string) {
+	if currentVersion == "dev" {
+		return
+	}
+	semVer := strings.Replace(currentVersion, "v", "", -1)
+	v := semver.MustParse(semVer)
+	latest, err := selfupdate.UpdateSelf(v, "gravitl/netmaker")
```

Comment:
Not sure but is this supposed to be "gravitl/netclient" rather?

Reply:
netclient release binaries are present in the netmaker repo https://github.com/gravitl/netmaker/releases/tag/v0.17.0

Reply:
they won't be in the future and the binaries in the netmaker repo will not be compatible with the file structure and the servers that binaries from this repo communicate with

Reply:
done